app/controllers/posts_controller.rb

```diff
@@ -20,9 +20,7 @@ def create
 
     respond_to do |format|
       if @post.save
-        format.html { redirect_to user_post_path(@post.author_id, @post), notice: 'Post was successfully created.' }
-      else
-        format.html { render :new, status: :unprocessable_entity }
+        format.html { redirect_to user_posts_path(current_user, @post), notice: 'Post was successfully created.' }
```

Comment:
- [ ] You have to use  **CanCanCan** for Post authorization in `Post Controller` as you did with your `Comments Controller`

   ```ruby
     class PostsController < ApplicationController
       load_and_authorize_resource
   
       ...
     end
   ```
- The load_and_authorize_resource method is provided to automatically authorize all actions in a RESTful style resource controller. It will use a before action to load the resource into an instance variable and authorize it for every action. Kindly add this 👍🏻

Reply:
Hie @BenMukebo I had put this initially but I removed it because it would not allow any user to add comments and like. So I had to configure it manually. If you run the project, the authorization works as it is supposed to.

Reply:
@tafaramafemba All depend on the way defines the set of rules for user inside the ability model 👇🏻 

```ruby
 # app/models/ability.rb

 class Ability
 include CanCan::Ability

     def initialize(user)

         user ||= User.new # guest user (not logged in)

         if user.admin? 
         can :manage, :all
         else
         can :manage, Post, author: user  # owner of post can perform all view, update or destroy own post if Post.author == user
         can :manage, Comment, author: user
         can :read, :all
         end
     end
 end
```